tests/client/client.rs

```diff
@@ -0,0 +1,21 @@
+use bson::Bson;
+use mongodb::client::MongoClient;
+
+#[test]
+fn list_databases() {
```

Comment:
On a second look at the python driver and the cmd reply, it looks like `list_databases` returns all databases in a single document without a cursor_id, so it should be run through `command` and return a `Vec<String>`. It may be more API consistent to return a cursor and provide a separate `database_names()` function, though, since `list_collections()` definitely provides some form of batching using a cursor.